bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -81,15 +81,29 @@ async def apply_infraction(
         ctx: Context,
         infraction: _utils.Infraction,
         user: UserSnowflake,
-        action_coro: t.Optional[t.Awaitable] = None
-    ) -> None:
-        """Apply an infraction to the user, log the infraction, and optionally notify the user."""
+        action_coro: t.Optional[t.Awaitable] = None,
+        reason_override: t.Optional[str] = None,
+        additional_info: t.Optional[str] = None,
+    ) -> bool:
+        """
+        Apply an infraction to the user, log the infraction, and optionally notify the user.
+
+        `reason_override`, if provided, will be sent to the user in place of the infraction reason.
+        `additional_info` will be attached to the text field in the mod-log embed.
+        Returns whether or not the infraction succeeded.
+        """
         infr_type = infraction["type"]
         icon = _utils.INFRACTION_ICONS[infr_type][0]
         reason = infraction["reason"]
```

Comment:
```suggestion
        reason = infraction["reason"] if reason_override is not None else reason_override
```
I think something along the lines of this would be a better way of setting the reason for the infraction.

Reply:
Perhaps the name `reason_override` is a bit misleading, it's for having a different message be sent to the user as the `reason` for the infraction rather than the reason sent to `#mod-log`. I've renamed the parameter to hopefully make it clearer.

Reply:
I agree that the new name is better.